src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -1090,6 +1090,45 @@ VideoDecoder::BatchDecodedOutput VideoDecoder::getFramesAtIndices(
   return output;
 }
 
+VideoDecoder::BatchDecodedOutput VideoDecoder::getFramesDisplayedByTimestamps(
+    int streamIndex,
+    const std::vector<double>& framePtss) {
+  validateUserProvidedStreamIndex(streamIndex);
```

Comment:
Maybe add a TODO saying long term we should not require scanning the file for time-based frame extraction